content/docs/reference/downstream-mtls-settings.mdx

```diff
@@ -0,0 +1,308 @@
+
+```yaml
+downstream_mtls:
+  enforcement: reject_connection
+```
+
+```bash
+DOWNSTREAM_MTLS_ENFORCEMENT=reject_connection
+```
+
+## Match Subject Alt Names {#match-sans}
+
+**Match Subject Alt Names** can be used to add an additional constraint when validating client certificates: each certificate must contain a Subject Alternative Name (SAN) of a particular type, whose value matches the provided regular expression.
+
+Client certificates satisfying _at least one_ of the provided match conditions will be allowed.
+
+Supported SAN types include:
+
+- `dns` — a DNS name
+- `email` — an email address (this is the `rfc822Name` as specified in [RFC 5280 §4.2.1.6](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6))
+- `ip_address` — an IP address (either IPv4 or IPv6)
+- `uri` — a Uniform Resource Identifier (URI)
+
+The provided regular expression must match against the entire SAN entry. The regular expressions use Google's [RE2 syntax](https://github.com/google/re2/wiki/Syntax).
+
+SAN IPv4 addresses will be expressed in dotted decimal notation (e.g. "127.0.0.1") when matching against the provided regular expression. SAN IPv6 addresses will be expressed as specified in [RFC 5952](https://datatracker.ietf.org/doc/html/rfc5952#section-4).
+
+### How to configure {#match-sans-how-to-configure}
+
+<Tabs groupId="deployment-type">
+<TabItem value="Core" label="Core">
+
+| **Config file keys** | **Environment variables** | **Type** |
+| :-- | :-- | :-- |
+| `downstream_mtls.match_subject_alt_names` | n/a | array of mappings from SAN type to regular expression |
+
+</TabItem>
+<TabItem value="Enterprise" label="Enterprise">
+
+This setting is not configurable in the Enterprise Console.
+
+</TabItem>
+<TabItem value="Kubernetes" label="Kubernetes">
+
+The Ingress Controller does not support this setting.
+
+</TabItem>
+</Tabs>
+
+### Examples {#match-sans-examples}
+
+To require that all client certificates have a DNS SAN with a domain name ending in ".corp.example.com":
+
+```yaml
+downstream_mtls:
+  match_subject_alt_names:
+    - dns: '.*\.corp\.example\.com'
+```
+
+To require that all client certificates have _either_ a DNS SAN containing the string "beyondcorp" _or_ an email address SAN with the domain "example.com":
+
+```yaml
+downstream_mtls:
+  match_subject_alt_names:
+    - dns: '.*beyondcorp.*'
+    - email: '.*@example\.com'
+```
+
+## Max Verify Depth
+
+**Max Verify Depth** sets a limit on the depth of a certificate chain presented by the client. The depth is defined as the number of certificates in the verified trust chain, _excluding_ the trust anchor. (The trust anchor can be any root or intermediate CA certificate included in the trusted [**CA**](#ca) bundle.)
+
+When set to 0, no specific limit is enforced.
+
+For compatibility with previous Pomerium releases, the default is 1. That is, only certificates _directly_ issued by a CA included in the trusted CA bundle are allowed.
+
+:::note
```

Comment:
Should this be caution since it _may_ break in the future?

https://www.notion.so/pomerium/Admonition-Levels-bb8b364880024af68cdeeff7fc658d44?pvs=4#5ea8101614bb4a128c8c3fd7c2954ce1

Reply:
Done.